app/containers/Tracks/index.js

```diff
@@ -68,104 +62,120 @@ const StyledOutlinedInput = styled(OutlinedInput)`
+    if (loaded) {
+      setLoading(false);
+    }
+  }, [tracksData]);
   useEffect(() => {
-      if (trackName && !tracksData?.results?.length) {
-        dispatchTrackDetails (trackName);
-        setLoading(true);
-      }
-    }, []);
-    
-  const searchTracks = (rName) => {
-      dispatchTrackDetails(rName);
+    if (trackName && !tracksData?.results?.length) {
+      dispatchTrackDetails(trackName);
       setLoading(true);
-    };
-
-    const handleOnChange = (rName) => {
-      if (!isEmpty(rName)) {
-        searchTracks(rName);
-      } else {
-        dispatchClearTrackDetails();
-      }
-    };
-
-    const handleStoriesClick = () => {
-      history.push('/stories');
-      window.location.reload();
-    };
-    const renderSkeleton = () => {
-      return (
-        <>
-          <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
-          <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
-          <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
-        </>
-      );
-    };
-
-    const viewTracks = (item) => {
-      let finalurl = '/tracks/'+item.trackId;
-      history.push({pathname:finalurl, state:{item:item}});
-      window.location.reload();
+    }
+  }, []);
 
+  const searchTracks = (trackName) => {
+    dispatchTrackDetails(trackName);
+    setLoading(true);
+  };
+
+  const handleOnChange = (rName) => {
+    if (!isEmpty(rName)) {
+      searchTracks(rName);
+    } else {
+      dispatchClearTrackDetails();
     }
+  };
 
-    const renderTrackList = () => {
-      
-      const items = get(tracksData, 'results', []);
-      const totalCount = get(tracksData, 'resultCount', 0);
-      return (
-        <If condition={!isEmpty(items) || loading}>
-          <CustomCard>
-            <If condition={!loading} otherwise={renderSkeleton()}>
-              <>
-               
-                <If condition={totalCount !== 0}>
-                  <div>
-                    <T id="records_found" values={{ totalCount }} />
-                  </div>
-                </If>
-                <div style={{display: 'grid', gridTemplateColumns: 'repeat(auto-fit, minmax(300px, 1fr))', gridGap: '1rem'}}>
+  const renderSkeleton = () => {
+    return (
+      <>
+        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
+        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
+        <Skeleton data-testid="skeleton" animation="wave" variant="text" height={40} />
+      </>
+    );
+  };
+
+  const viewTracks = (item) => {
+    let finalurl = '/tracks/' + item.trackId;
+    history.push({ pathname: finalurl, state: { item: item } });
+    window.location.reload();
+  };
+
+  const renderTrackList = () => {
+    const items = get(tracksData, 'results', []);
+    const totalCount = get(tracksData, 'resultCount', 0);
+    return (
+      <If condition={!isEmpty(items) || loading}>
```

Comment:
Do not define your components in the Container. Instead use component folder.
In this file I can see 3 renders  renderSkeleton, renderTracklist and renderErrorState. Lets move all their respective html code to separate components in the component folder like <Skeleton > <TrackList > and <ErrorState>